src/SegmentedControl.tsx

```diff
@@ -0,0 +1,198 @@
+import {
+    CSSProperties,
+    DetailedHTMLProps,
+    forwardRef,
+    InputHTMLAttributes,
+    memo,
+    ReactNode,
+    useId
+} from "react";
+import { assert, Equals } from "tsafe";
+import { fr, FrIconClassName, RiIconClassName } from "./fr";
+import React from "react";
+import { CxArg } from "tss-react";
+import { cx } from "./tools/cx";
+import { useAnalyticsId } from "./tools/useAnalyticsId";
+
+export type SegmentedControlProps = {
+    id?: string;
+    className?: string;
+    name?: string;
+    classes?: Partial<
+        Record<
+            "root" | "legend" | "hintText" | "elements" | "element-each" | "element-each__label",
+            CxArg
+        >
+    >;
+    style?: CSSProperties;
+    /** default: false */
+    small?: boolean;
+    /**
+     * Minimum 1, Maximum 5.
+     *
+     * All with icon or all without icon.
+     */
+    segments: SegmentedControlProps.Segments;
+} & (
+    | SegmentedControlProps.WithLegend
+    | SegmentedControlProps.WithInlineLegend
+    | SegmentedControlProps.WithHiddenLegend
+);
+
+//https://main--ds-gouv.netlify.app/example/component/segmented/
+export namespace SegmentedControlProps {
+    export type WithLegend = {
+        inlineLegend?: boolean;
+        legend: ReactNode;
+        hintText?: ReactNode;
+        hideLegend?: boolean;
+    };
+
+    export type WithInlineLegend = {
+        inlineLegend: true;
+        legend: ReactNode;
+        hintText?: ReactNode;
+        hideLegend?: never;
+    };
+
+    export type WithHiddenLegend = {
+        inlineLegend?: never;
+        legend?: ReactNode;
+        hintText?: never;
+        hideLegend: true;
+    };
+
+    export type Segment = {
+        label: ReactNode;
+        nativeInputProps?: DetailedHTMLProps<
+            InputHTMLAttributes<HTMLInputElement>,
+            HTMLInputElement
+        >;
```

Comment:
Same here, I think it's cleaner to use `React.ComponentProps<"input">` (I wasn't aware of this utility type before that's why it's not used in the other components 😄 )